test/AutoRest.TestServer.Tests/PublicClientCtorTests.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using AutoRest.TestServer.Tests.Infrastructure;
+using Azure;
+using Azure.Core;
+using NUnit.Framework;
+using PublicClientCtor;
+
+namespace AutoRest.TestServer.Tests
+{
+    public class PublicClientCtorTests : TestServerTestBase
+    {
```

Comment:
Let's add tests that verify that intended headers are being sent.
You can use https://github.com/Azure/autorest.csharp/blob/feature/v3/test/AutoRest.TestServer.Tests/HeaderCollectionPrefixTests.cs#L16 as a sample of how to host and inprocess server.